src/control/server/server_utils.go

```diff
@@ -199,10 +219,18 @@ func prepBdevStorage(srv *server, iommuEnabled bool) error {
 	}
 
 	if hasBdevs {
+		// The NrHugepages config value is a total for all engines. Distribute allocation
+		// of hugepages equally across each engine's numa node (as validation ensures that
+		// TargetsCount is equal for each engine).
+		numaNodes := getEngineNUMANodes(srv.log, srv.cfg.Engines)
+
 		// Request a few more hugepages than actually required as not all may be available.
 		prepReq.HugePageCount = (srv.cfg.NrHugepages + common.ExtraHugePages)
+		prepReq.HugePageCount /= len(numaNodes)
```

Comment:
I don't fully understand how the hugepage allocation works under the covers. If we're allocating them per NUMA node, do we want to tack on the extra pages to each request, rather than dividing them up amongst the request? I guess the question is whether it's still enough padding if we divide them up.

Reply:
done